### PR TITLE
[Refactor] User authentication use-case

### PR DESCRIPTION
**Pull request by youssefsoli**
This PR fixes a few issues after PR https://github.com/CSC207-2022F-UofT/course-project-fitbook/pull/23.

[Renamed userName to name](https://github.com/CSC207-2022F-UofT/course-project-fitbook/commit/9a615f9f8dbea0971ca24ea284431888524a0280)
  
[Added username search related exceptions](https://github.com/CSC207-2022F-UofT/course-project-fitbook/commit/d0a8b5f98519282831d3fce0dff35baf277a7f03)
  
[Implemented user search related features to the firebase repository](https://github.com/CSC207-2022F-UofT/course-project-fitbook/commit/32f97255dc6a3f3002b59403807718aacf03179e)
  
[Fixed != vs !.equals for string comparison bug](https://github.com/CSC207-2022F-UofT/course-project-fitbook/commit/e45741f29fe9db0eee48bbfe45dc426b2c951818)
  
[Added static keyword to local exceptions](https://github.com/CSC207-2022F-UofT/course-project-fitbook/commit/61ae84d3cc17d5d54c1d1e5ca42d05b8cfb56ca0)
  
[Added comments and a check for short usernames](https://github.com/CSC207-2022F-UofT/course-project-fitbook/commit/2b58320cb309e9c5113ac7fdb1d9cac9f31314c6)

### CODE REVIEW

**Review by modaserMoj — APPROVED** [review]
Improved the code for the login and signup use cases by adding comments and optimizing the code.

File: src/main/java/ca/utoronto/fitbook/application/port/in/UsernameNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package ca.utoronto.fitbook.application.port.in;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR, reason = "Username collision.")
```
Comment (Ludosss) [inline src/main/java/ca/utoronto/fitbook/application/port/in/UsernameNotFoundException.java:6]: change HTTP response code (422) and reason ("Username not found")

**Review by Ludosss — CHANGES_REQUESTED** [review]

**Review by modaserMoj — APPROVED** [review]

**Review by Ludosss — APPROVED** [review]
looks good

**Review by jpablo2002 — APPROVED** [review]
Lgtm

**Review by JacketPants12 — APPROVED** [review]
lgtm

**Pull request merged**